test/test_nn.py

```diff
@@ -4470,6 +4470,37 @@ def test_load_state_dict_ref_cycle(self):
 
         self.assertEqual(refcycles, 0)
 
+    def test_load_state_dict_custom(self):
+
+        class CustomState(nn.Module):
+            def __init__(self):
+                super(CustomState, self).__init__()
+                self.param = torch.nn.Parameter(torch.ones(1))
+                self.sub = torch.nn.Linear(5,5)
+
+            def _save_to_state_dict(self, destination, prefix, keep_vars):
+                destination[prefix + "serialized"] = self.param.data + 1
+
+            def _load_from_state_dict(self, state_dict, prefix, local_metadata,
+                                      strict, missing_keys, unexpected_keys,
+                                      error_msgs):
+                # skip some of the error handling
+                self.param.data.copy_(state_dict[prefix + "serialized"] - 1)
+
+        m = CustomState()
+        m.param[0] = 10
+        m.sub.weight[0, 0] = 555
+        state_dict = m.state_dict()
+        self.assertEqual(state_dict["serialized"].item(), 11)
+        self.assertIn("sub.weight", state_dict)
+        self.assertNotIn("param", state_dict)
+        del m
+        mm = CustomState()
+        self.assertEqual(mm.param[0].item(), 1)
+        mm.load_state_dict(state_dict)
+        self.assertEqual(mm.param[0].item(), 10)
+        self.assertEqual(mm.sub.weight[0, 0].item(), 555)
```

Comment:
Though I suppose you can appease soumith by testing the nested case explicitly here ;)